torch/csrc/dynamo/eval_frame.c

```diff
@@ -71,44 +72,12 @@ static struct PyGetSetDef THPDevice_properties[] = {
     {"f_lasti", (getter)THPPyInterpreterFrame_f_lasti, NULL, NULL, NULL},
     {NULL}};
 
-PyTypeObject THPPyInterpreterFrameType = {
-    PyVarObject_HEAD_INIT(NULL, 0) "torch._C.dynamo.eval_frame._PyInterpreterFrame", /* tp_name */
-    sizeof(THPPyInterpreterFrame), /* tp_basicsize */
-    0, /* tp_itemsize */
-    NULL, /* tp_dealloc */
-    0, /* tp_vectorcall_offset */
-    NULL, /* tp_getattr */
-    NULL, /* tp_setattr */
-    NULL, /* tp_reserved */
-    NULL, /* tp_repr */
-    NULL, /* tp_as_number */
-    NULL, /* tp_as_sequence */
-    NULL, /* tp_as_mapping */
-    NULL, /* tp_hash  */
-    NULL, /* tp_call */
-    NULL, /* tp_str */
-    NULL, /* tp_getattro */
-    NULL, /* tp_setattro */
-    NULL, /* tp_as_buffer */
-    Py_TPFLAGS_DEFAULT, /* tp_flags */
-    NULL, /* tp_doc */
-    NULL, /* tp_traverse */
-    NULL, /* tp_clear */
-    NULL, /* tp_richcompare */
-    0, /* tp_weaklistoffset */
-    NULL, /* tp_iter */
-    NULL, /* tp_iternext */
-    NULL, /* tp_methods */
-    NULL, /* tp_members */
-    THPDevice_properties, /* tp_getset */
-    NULL, /* tp_base */
-    NULL, /* tp_dict */
-    NULL, /* tp_descr_get */
-    NULL, /* tp_descr_set */
-    0, /* tp_dictoffset */
-    NULL, /* tp_init */
-    NULL, /* tp_alloc */
-    NULL, /* tp_new */
+static PyTypeObject THPPyInterpreterFrameType = {
```

Comment:
I always copy-paste this one from existing code. I guess your take is that we should use the short version going forward?

Reply:
Designated initializers (this syntax) are in C99 and in C++20 (but not C++17). This is a C file, so that's fine, but I don't think we use C++20 yet. It's really minor, but I tend to prefer it because, I think it's easier to read, less likely to have bugs due to mismatching field names, and a bit less likely to break in future releases due to CPython changing details of the struct (most fields are unused and those are the ones most frequently changed).

Reply:
Ok!
Sounds good even though that might prevent us from moving that file to C++ if we want to.